fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/ReindexService.java

```diff
@@ -209,6 +209,54 @@ public void reset() {
         membershipService.reset();
     }
 
+    /**
+     * Commit the records added from transaction.
+     * @param transaction the transaction.
+     */
+    public void commit(final Transaction transaction) {
+        try {
+            LOGGER.debug("Performing commit of transaction {}", transaction);
+            containmentIndex.commitTransaction(transaction);
+            ocflIndex.commit(transaction);
+            referenceService.commitTransaction(transaction);
+            searchIndex.commitTransaction(transaction);
+            LOGGER.debug("Finished commit of transaction {}", transaction);
+        } catch (final RuntimeException e) {
+            rollback(transaction);
+            throw e;
+        }
+    }
+
+    /**
+     * Quietly rollback all changes to the various indexes.
+     * @param transaction the transaction to rollback.
+     */
+    public void rollback(final Transaction transaction) {
+        execQuietly("Failed to reset searchIndex", () -> {
+            searchIndex.rollbackTransaction(transaction);
+            return null;
+        });
+
+        execQuietly("Failed to rollback containment index transaction " + transaction, () -> {
+            containmentIndex.rollbackTransaction(transaction);
+            return null;
+        });
+        execQuietly("Failed to rollback OCFL index transaction " + transaction, () -> {
+            ocflIndex.rollback(transaction);
+            return null;
+        });
+
+        execQuietly("Failed to rollback the reference index transaction " + transaction, () -> {
+            referenceService.rollbackTransaction(transaction);
+            return null;
+        });
+
+        execQuietly("Failed to rollback membership index transaction " + transaction, () -> {
+            membershipService.rollbackTransaction(transaction);
+            return null;
+        });
+    }
+
```

Comment:
The changes to the `ReindexService` are not needed because this is all handled within the tx now

Reply:
okay - I'll remove.